hircluster.c

```diff
@@ -1383,30 +1383,33 @@ static int cluster_update_route_by_addr(redisClusterContext *cc, const char *ip,
         }
     }
 
-    // Move all hiredis contexts in cc->nodes to nodes
-    cluster_nodes_swap_ctx(cc->nodes, nodes);
-
-    /* Replace cc->nodes before releasing the old dict since
-     * the release procedure might access cc->nodes. */
-    oldnodes = cc->nodes;
-    cc->nodes = nodes;
-    if (oldnodes != NULL) {
-        dictRelease(oldnodes);
-    }
-
     if (cc->slots != NULL) {
         cc->slots->nelem = 0;
         hiarray_destroy(cc->slots);
     }
     cc->slots = slots;
 
+    /* Update slot-to-node table before changing cc->nodes since
+     * removal of nodes might trigger user callbacks.
+     * Required if commands are sent from these callbacks. */
```

Comment:
In the second sentence, it's slightly unclear what is required. How about something like this?

```suggestion
    /* Update slot-to-node table before changing cc->nodes since
     * removal of nodes might trigger user callbacks which may
     * send commands, which depend on the slot-to-node table. */
```